### PR TITLE
DB-12131 Close result set in case of exception 

### PR DESCRIPTION
**Pull request by arnaud-splice**
## Description

Close result set if insert operation went wrong

## How to test

No particular tests to run on top of our usual integration tests.

## All DB-12131 PRs

[spliceengine master](https://github.com/splicemachine/spliceengine/pull/5636)
[spliceengine branch-3.0](https://github.com/splicemachine/spliceengine/pull/5637)
[spliceengine branch-3.1](https://github.com/splicemachine/spliceengine/pull/5638)

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Review by dgomezferro — APPROVED** [review]

File: hbase_sql/src/main/java/com/splicemachine/mrio/api/hive/SMHiveRecordWriter.java [hunk 1]
```diff
@@ -87,6 +87,13 @@ public void write(RowLocationWritable key, ExecRowWritable value)
             insertOperation.open();
             insertOperation.close();
         } catch (StandardException e) {
+            if (insertOperation != null) {
+                try {
+                    insertOperation.close();
+                } catch (StandardException ex) {
+                    LOG.error(e);
+                }
+            }
```
Comment (ascend1) [inline hbase_sql/src/main/java/com/splicemachine/mrio/api/hive/SMHiveRecordWriter.java:96]: It looks like `insertOperation.close()` should be in a `finally` block. If line 88 throws, line 92 would throw again (or some other behavior).

**Review by ascend1 — APPROVED** [review]
With a minor comment.

File: hbase_sql/src/main/java/com/splicemachine/mrio/api/hive/SMHiveRecordWriter.java (hunk 1 above)
Comment (arnaud-splice) [reply hbase_sql/src/main/java/com/splicemachine/mrio/api/hive/SMHiveRecordWriter.java]: Done

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3288/

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**